### ISSUE TITLE
False negative on `ForbiddenMethodCall`

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#8786]
## Expected Behavior
```kt
@Test
fun `should report property getters call used without property`() {
    val code = """
        fun foo() = run {
            val calendar = java.util.Calendar.getInstance()
            calendar.firstDayOfWeek
        }
    """.trimIndent()
    val findings = ForbiddenMethodCall(
        TestConfig(METHODS to listOf("java.util.Calendar.getFirstDayOfWeek"))
    ).lintWithContext(env, code)
    assertThat(findings).hasSize(1)
}
```

above TC should pass

## Observed Behavior
Above TC is failing

## Steps to Reproduce
This is happening in the latest main branch(98e26b62f5a7aae642d30bff1124d2f2d47e8d15)

## Your Environment

* Version of detekt used: Latest main branch(98e26b62f5a7aae642d30bff1124d2f2d47e8d15)
